modin/data_management/utils.py

```diff
@@ -59,6 +59,8 @@ def compute_chunksize(df, num_splits, default_block_size=32, axis=None):
         If axis is 1 or 0, returns an integer number of rows/columns to split the
         DataFrame. If axis is None, returns a tuple containing both.
     """
+    # `num_splits` should be a non-zero value
+    num_splits = num_splits if num_splits > 0 else 1
```

Comment:
`num_splits` is also used in functions:
* `_define_metadata`;
* `split_result_of_axis_func_pandas`;
* Based on this variable, a `column_widths` variable for `build_partition` function is determined (For example, in `JSONDispatcher._read`)

Should we also change these places for consistency?

It would be nice to add examples to some other I/O functions (For example, `read_json`).

Reply:
In the case of empty frame reading, `num_splits` will be redefined to 1 in the `_define_metadata` and then processed normally further, but i think it will be more correct to set correct `num_splits` value in the moment of `num_splits` definition, so previous approach was used.
I suggest to fix in this PR only `read_csv` function as it mentioned in the issue. Actually, what you mean under "examples"?

Reply:
Oh, I wanted to say `add more IO tests` :)

Reply:
There are a few more places where similar lines can be found:
* https://github.com/modin-project/modin/blob/54898ef78e83209fba3233606841ca988e531aee/modin/engines/base/io/text/excel_dispatcher.py#L163
* https://github.com/modin-project/modin/blob/54898ef78e83209fba3233606841ca988e531aee/modin/engines/base/io/text/fwf_dispatcher.py#L157

I believe that the same error exists for them. Why make changes for them separately if the problematic mechanism (`num_splits` definition) is the same?

Reply:
Yes, you are right, the same issue can occur for another dispatchers, but for excel dispatcher we wouldn't be capable to verify this fix because of https://github.com/modin-project/modin/issues/3305, so additional PR should be done anyway. But, i think, it should be fine to add these fixes anyway, so they were added.